### ISSUE TITLE
Add support for MCT tool 's dump format...

### ISSUE THREAD

**Issue by iceman1001** [iceman1001/ChameleonMini-rebootedGUI#90]
The MCT tool has its own format,  it would be nice if the GUI could support it aswell

https://github.com/ikarus23/MifareClassicTool/issues/217

@kgamecarter    Maybe your magic touch?

**Comment by kgamecarter** [comment]
Ok
MCT is a nice tool for mobile read/write card.
I also want to support the MCT format.

**Comment by iceman1001** [comment]
@kgamecarter  Great addition!    Now I only need a MCT dump file to test/verify it with  ( @ikarus23   ?)

**Comment by iceman1001** [comment]
Converting back and forward looks great.
...may add a refresh button,  right-click,   or after a convertion to update the file explorer...

**Comment by iceman1001** [comment]
whoops..  converting a Ultralight bin dump to EML or MCT ..  will try to make a classic  formatted....

**Comment by kgamecarter** [comment]
MCT is a tool only for classic , so just only classic format.

Eml convertion code is copied from https://github.com/iceman1001/ChameleonMini-rebootedGUI/blob/1a345a9ee73312745cc4908b50945ed7142cd455/ChameleonMiniGUI/UserControls/UcExplorer.cs#L301
Does Ultralight also have EML format?
If so. I need a Ultralight EML sample.

**Comment by iceman1001** [comment]
:)  the UL/NTAG dumps are more complex in one sense.   I have bin for it,  the json is not done yet...
...if you don't want to add json in pm3 client for **hf mfu dump**

**Comment by kgamecarter** [comment]
ok
it's 4 byte per line.

**Comment by iceman1001** [comment]
...the old dumps was 4byte, but then we added a block of special data in the beginning of the dump file

**Comment by iceman1001** [comment]
https://github.com/RfidResearchGroup/proxmark3/blob/master/client/cmdhfmfu.c#L1856

**Comment by kgamecarter** [comment]
The special data does not seem to match EML format.
needs a better format for mfu_dump_t

**Comment by iceman1001** [comment]
I know...   hence the urge to get JSON for it :)

**Comment by iceman1001** [comment]
Alrighty then.    **hf mfu dump** new json dump format in proxmark3 client.

https://pastebin.com/gn5wG3KM

**Comment by iceman1001** [comment]
as of this moment I don't think the Chameleon uses the special data.  It really should though.
Better look in the firmware repo.  Or @bogiton , do you know ?

As of now,  I think it can only load / convert JSON format. not save it.

**Comment by iceman1001** [comment]
How about we just set those special fields as empty?

**Comment by bogiton** [comment]
The only extra thing we save are the counters for UL, right after the memory blocks. I had a check added for this in the DownloadAndSaveDump method.

**Comment by iceman1001** [comment]
But do we have access to Version, Signature etc?

**Comment by bogiton** [comment]
Nope, just the memory block bytes. Signature is hard-coded I think.

**Comment by kgamecarter** [comment]
The binary dump only end of user memory?
No include lock bytes and configurations page?
![img](https://i.imgur.com/Ni0F9tP.png)

**Comment by iceman1001** [comment]
There are so many different variants of Ultralight / NTAG etc class tags so keeping track of each cards memory layout is hard. Not to mention locking mechanisms. The pm3 client dumps all blocks it can read.

**Comment by iceman1001** [comment]
if pm3 client can figure out default pwd/pack,   it adds those for known tag formats.  Not all formats.
